libdnf/system/state.hpp

```diff
@@ -223,6 +225,17 @@ class State {
     /// @since 5.0
     void reset_module_states(std::map<std::string, ModuleState> new_states) { module_states = new_states; }
 
+    /// Reset packages system state to match given values.
+    /// @param installed_packages Vector of tuples <rpm::Nevra nevra, TransactionItemReason reason, std::string repository_id> of currently installed packages
+    /// @param installed_groups Vector of tuples <std::string group_id, TransactionItemReason reason, std::set<std::string> installed_packages> of currently installed groups
+    /// @param installed_environments Vector of tuples <std::string environment_id, std::set<std::string> installed_groups> of currently installed environmental groups
+    /// @since 5.0
+    void reset_packages_states(
+        std::map<std::string, libdnf::system::PackageState> package_states,
+        std::map<std::string, libdnf::system::NevraState> nevra_states,
+        std::map<std::string, libdnf::system::GroupState> group_states,
+        std::map<std::string, libdnf::system::EnvironmentState> environment_states);
```

Comment:
I am not sure whether it is intended. Argument `package_states` creates a copy of map and then move it to `this->package_states = std::move(package_states);`. In the last commit there is -
```
            system_state.reset_packages_states(
                std::move(package_states),
                std::move(nevra_states),
                std::move(group_states),
                std::move(environment_states));
        }
```
I suggest that you meant to use `std::map<std::string, libdnf::system::PackageState> && package_states`. Then the code call can be without std::move()
```
            system_state.reset_packages_states(
                package_states,
                nevra_states,
                group_states,
                environment_states);
        }
```
May be it would be also nice to state that methods take ownership of arguments.

Reply:
You cannot use rvalue reference argument and than use the function with lvalue. But I've changed the `reset_packages_states()` signature to use rvalue references.